lib/blocks.php

```diff
@@ -431,4 +431,72 @@ function gutenberg_legacy_wp_block_post_meta( $value, $object_id, $meta_key, $si
 
 	return $value;
 }
+
 add_filter( 'default_post_metadata', 'gutenberg_legacy_wp_block_post_meta', 10, 4 );
+
+
+/**
+ * Check if the lightbox should be rendered for a given block.
+ *
+ * This function is INTENTIONALLY left out of core as it only provides
+ * backwards compatibility for the legacy lightbox syntax that was only
+ * introduced in Gutenberg. The legacy syntax was using the `behaviors` key in
+ * the block attrbutes and the `theme.json` file.
+ *
+ * @param array $block The block to check.
+ * @return array The block with the lightboxEnabled flag set if the lightbox should be rendered.
+ */
+function gutenberg_should_render_lightbox( $block ) {
+
+	if ( 'core/image' !== $block['blockName'] ) {
+		return $block;
+	}
+
+	// Get the lightbox setting from the block attributes.
+	if ( isset( $block['attrs']['lightbox'] ) ) {
+		$lightbox_settings = $block['attrs']['lightbox'];
+		// If not present, check legacy syntax.
+	} elseif ( isset( $block['attrs']['behaviors']['lightbox'] ) ) {
+		$lightbox_settings = $block['attrs']['behaviors']['lightbox'];
+	}
```

Comment:
> The correct way to handle attribute migration is by adding a [block deprecation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/).

> I'd suggest that you also keep around the lib/block-supports/behaviors.php file and add a [_deprecated_function()](https://developer.wordpress.org/reference/functions/_deprecated_function) call just inside [gutenberg_render_behaviors_support_lightbox](https://github.com/WordPress/gutenberg/pull/53851/files#diff-b2dcc99da76a0f884845c2754c15d331e85d1da19e6246858ba2b37bb068f119L46).

Do you mean that we should remove this logic and retain the [gutenberg_render_behaviors_support_lightbox](https://github.com/WordPress/gutenberg/pull/53851/files#diff-b2dcc99da76a0f884845c2754c15d331e85d1da19e6246858ba2b37bb068f119L46) for backwards compatibility instead?

> This way, if someone has already started using the block support in one of their blocks, they can be warned to add a block deprecation. Be sure to add a dev note explaining how since this is a rather unusual deprecation.

`Behaviors` was only ever supported for the image block, and only for the lightbox. Do you think this is still necessary? Tagging @gziolo for input as well.

Reply:
> Do you mean that we should remove this logic and retain the [gutenberg_render_behaviors_support_lightbox](https://github.com/WordPress/gutenberg/pull/53851/files#diff-b2dcc99da76a0f884845c2754c15d331e85d1da19e6246858ba2b37bb068f119L46) to support backwards compatibility instead?
>
> `Behaviors` was only ever supported for the image block, and only for the lightbox. Do you think this is still necessary?

Yes and yes. The block support could have been added by a 3rd party block, so we don't know if the image block was the only block to use the block support. Since the block support never made it into core, we can still get rid of it in two versions, but we need to add a deprecation notice for those two versions that warns users of the support that it will be going away and they will need to do a manual migration (like you did for the image block) if they'd like to keep using it.

Reply:
> I'd suggest that you also keep around the lib/block-supports/behaviors.php file and add a [_deprecated_function()](https://developer.wordpress.org/reference/functions/_deprecated_function) call just inside [gutenberg_render_behaviors_support_lightbox](https://github.com/WordPress/gutenberg/pull/53851/files#diff-b2dcc99da76a0f884845c2754c15d331e85d1da19e6246858ba2b37bb068f119L46).

I've added `_deprecated _function` as you've suggested as well as revised the backwards compatibility in https://github.com/WordPress/gutenberg/pull/54071/commits/0e63890cb1415266bdd292c5e477bc99b0d9595c